navigator/backend/src/main/scala/com/digitalasset/navigator/store/platform/PlatformStore.scala

```diff
@@ -159,37 +162,72 @@ class PlatformStore(
+        )
+        .andThen {
+          case Success(_) => // .. then only list users on the login screen
+            state.ledgerClient.userManagementClient
+              .listUsers() // don't pass token here -- it's already set on startup (LedgerClientConfiguration)
+              .map(UpdatedUsers(_))
+              .pipeTo(self)
+          case Failure(_) => // ... else fallback to parties
+            state.ledgerClient.partyManagementClient
+              .listKnownParties()
+              .map(UpdatedParties(_))
+              .pipeTo(self)
+        }
+
       ()
+
+    case UpdatedUsers(users) =>
+      // Note: you cannot log in as a user without a primary party
+      val usersWithPrimaryParties = users.flatMap { user =>
+        user.primaryParty.map(p => (user.id, p))
+      }
+
+      usersWithPrimaryParties.foreach { case (userId, party) =>
+        self ! Subscribe(userId, new PartyState(ApiTypes.Party(party)))
+      }
+
     case UpdatedParties(details) =>
       details.foreach { partyDetails =>
         if (partyDetails.isLocal) {
-          val displayName = partyDetails.displayName match {
-            case Some(value) => value
-            case None => partyDetails.party
-          }
-          self ! Subscribe(
-            displayName,
-            UserConfig(party = ApiTypes.Party(partyDetails.party), role = None, useDatabase = false),
-          )
+          val displayName = partyDetails.displayName.getOrElse(partyDetails.party)
+          self ! Subscribe(displayName, new PartyState(ApiTypes.Party(partyDetails.party)))
         } else {
           log.debug(s"Ignoring non-local party ${partyDetails.party}")
         }
       }
 
-    case Subscribe(displayName, config) =>
+    case Subscribe(displayName, partyState) =>
       if (!state.parties.contains(displayName)) {
-        log.info(s"Starting actor for $displayName")
-        val partyState = new PartyState(config)
-        startPartyActor(state.ledgerClient, partyState)
-        context.become(connected(state.copy(parties = state.parties + (displayName -> partyState))))
+        log.info(s"Starting actor for ${partyState.name} (aka $displayName)")
+
+        // start party actor if needed (since users subscribe to their primary party,
+        // we may subscribe to the same party under different display names, but we should only create one actor per party)
+        val partyActorName = partyState.actorName
+        if (context.child(partyActorName).isEmpty)
+          context.actorOf(
+            PlatformSubscriber.props(state.ledgerClient, partyState, applicationId, token),
+            partyActorName,
+          )
+
+        val updatedParties = state.parties + (displayName -> partyState)
+        context.become(connected(state.copy(parties = updatedParties)))
       } else {
-        log.debug(s"Actor for $displayName is already running")
+        log.debug(s"Actor for ${partyState.name} (aka $displayName) is already running")
```

Comment:
```suggestion
        log.debug(s"Actor for party '${partyState.name}' (display name: '$displayName') is already running")
```
Probably worth being explicit here to avoid confusion.

Reply:
Furthermore, not necessarily for this PR, but what kind of logging is this? Should we rather use built-in lazy string interpolation rather then Scala's strict one?

Reply:
Something to revisit when this becomes a problem.